libs/@blockprotocol/graph/src/types/temporal-versioning.ts

```diff
@@ -45,6 +45,10 @@ export type Unbounded = { kind: "unbounded" };
  */
 export type TemporalBound = Unbounded | LimitedTemporalBound;
 
+/**
+ * A range of time from a given `start` {@link TemporalBound} (or lack of) to a given `end` {@link TemporalBound} (or
+ * lack of), where `start` is strictly before or equal to `end`.
+ */
 export type TimeInterval<
```

Comment:
I have tried to take this on board and applied it to a few things in the aim of consistency. Please see
- [Rename TimeInterval types](https://github.com/blockprotocol/blockprotocol/pull/928/commits/6dd35285f2ea2f0cfb68b3fb71573e83f4e4c3c9)
- [Rename Axis types](https://github.com/blockprotocol/blockprotocol/pull/928/commits/5a6b5105a4c2bef14c68c2691de3500770841273)
- [Rename Axes types](https://github.com/blockprotocol/blockprotocol/pull/928/commits/3fec1ea62d6e875f946f729c53efd0b106bdb897)

Happy to iterate on these until it makes more sense. I opted to try and centre on `Unresolved` rather than `Input`, but can also be swayed the other way. I've also put the `Unresolved` at the latter end of the type names now, which seemed more consistent if we avoid `Resolved` on the name.

Reply:
Had a look and dropped a comment. 

Personally I would not derive `TimeInterval` from `TimeIntervalUnresolved`. I don't think avoiding specifying `start` and `end` twice is worth the conceptual weirdness of saying that `TimeInterval` is a variant of  `TimeIntervalUnresolved`

Reply:
I thought about this too.. However, I _**really**_ dislike defining sub/super types completely separately in TypeScript, it makes me very uncomfortable and has lead to too many bugs in even my recent work.

That's why a while ago we ended up discussing the `Subtype` helper you found and that I've used in some places.

I could redefine `TimeIntervalUnresolved` as an `Omit` and replacement thing I guess...

Reply:
> I could redefine TimeIntervalUnresolved as an Omit and replacement thing I guess...

I think this would be better – in my head that's the right way round. `TimeIntervalUnresolved` is a special variation of `TimeInterval`, not vice versa (and you don't want people worrying about it when dealing with the normal `TimeInterval` / seeing it lurking at the base of the type)

Reply:
Except it's not a variation, it's not actually a sub-type 🙃 

I'll play around with it a bit

Reply:
It was all vile, opted for the simple thing (which I still really don't like 😭)

[Make TimeInterval independent of TimeIntervalUnresolved](https://github.com/blockprotocol/blockprotocol/pull/928/commits/619b4219291125078e38bb9f68a1be20961c8316)

Reply:
We can always change again later without affecting users – looks good for now!